src/Parser.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    /**
+     * @var path the file path to the csv file to be parsed.
+     */
+    private $path;
+
+    /**
+     * @var headers The column headers for the CSV file that give context to the column values.
+     */
+    private $headers;
+
+    /**
+     * @var delimiter the delimiter value that separates column values for this CSV file.
+     */
+    private $delimiter;
+
+    function __construct ($path, $delimiter = ',', $headers = []) 
+    {
+        /*
+         * Check that the path does exist or throws an exception
+         * Checks that the delimiter is a , | or /t
+         * Fills headers, if it's empty grabs first line from CSV instead
+         */
+
+        if(!file_exists($path)) 
+            throw new \Exception("{$path} is not a valid file.");
+        $this->path = $path; //fill path

```

Comment:
You should use fopen here so the Parser object keeps the file open while the object is active. This will be needed later when we start parsing rows of data since each time the file is opened `fgetcsv()` will have to start the from the beginning of the file which would be inefficient.

Reply:
This has been edited, not sure when the fclose should be used if it's even needed.